cache/refs.go

```diff
@@ -333,15 +303,20 @@ func (sr *immutableRef) Parent() ImmutableRef {
 	return nil
 }
 
-func (sr *immutableRef) ociDesc(ctx context.Context, dhs DescHandlers) (ocispecs.Descriptor, error) {
+func (cr *cacheRecord) ociDesc(ctx context.Context, dhs DescHandlers) (ocispecs.Descriptor, error) {
```

Comment:
fwiw I'm confused about a some of these reactors to cacheRecord. Immutableref is safe to call because calling it guarantees object is reference counted and locked.  `cacheRecord` does none of that and you need to follow to whole call chain to understand if these calls are safe. If there is no sharing of function code in here then I don't see the benefit.  Also functions for different types are mixed now. Not sure why linter doesn't detect it. Don't fix the order in here for easier review but eventually this should be done in separate PR.